peer.go

```diff
@@ -755,6 +815,15 @@ func ContextPeer(ctx context.Context) *Peer {
 	return nil
 }
 
+type execContextKey struct{}
```

Comment:
don't you want this to be a var of a single value? That's how I've wielded it at least, I thought different copies of a type would lead to different context slots?...

Reply:
I'm exactly the opposite of correct, disregard :sweat_smile: